### PR TITLE
Kibrantn/servicebus/track2 initial test cleanup

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
Initial overhaul of test structure.
Removes test blocks not present in track1. (topics, subscriptions)
Converts sync tests to track2 api.
fixes some minor bugs found during this process (reconnect, renew locks)
reads autorenew

### CODE REVIEW

File: sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py [hunk 1]
```diff
@@ -312,3 +312,12 @@ def close(self, exception=None):
...
         self._close_handler()
+
+    def reconnect(self):
+        """Reconnect the handler.
+
+        If the handler was disconnected from the service with
+        a retryable error, attempt to reconnect.
+        """
+        self.close()
+        self._open()
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_base_handler.py]: I think this reconnect can be removed, we're doing internal reconnect now.

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py [hunk 2]
```diff
@@ -181,7 +182,9 @@ def _create_handler(self, auth):
                 error_policy=self._error_policy,
                 client_name=self._name,
                 auto_complete=False,
-                encoding=self._config.encoding
+                encoding=self._config.encoding,
+                receive_settle_mode=self._mode.value,
+                timeout=self._idle_timeout * 1000 if self._idle_timeout else 0
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py]: I don't see self._idle_timeout attribute in the `__init__`, is it accidentally removed when merging?

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py [hunk 3]
```diff
@@ -447,3 +452,10 @@ def peek(self, message_count=1, sequence_number=None):
             message,
             mgmt_handlers.peek_op
         )
+
+    def _renew_locks(self, *lock_tokens):
+        message = {'lock-tokens': types.AMQPArray(lock_tokens)}
+        return self._mgmt_request_response(
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py]: _mgmt_request_response_with_retry

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py [hunk 4]
```diff
@@ -393,3 +395,10 @@ def from_connection_string(
             message,
             mgmt_handlers.peek_op
         )
+
+    async def _renew_locks(self, *lock_tokens):
+        message = {'lock-tokens': types.AMQPArray(lock_tokens)}
+        return await self._mgmt_request_response(
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/azure/servicebus/aio/_servicebus_receiver_async.py]: _with_retry

File: sdk/servicebus/azure-servicebus/tests/test_sb_client.py [hunk 5]
```diff
@@ -13,8 +13,8 @@
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/tests/test_sb_client.py]: Do we have test_sb_client_async?

File: sdk/servicebus/azure-servicebus/tests/test_queues.py [hunk 6]
```diff
@@ -72,75 +73,76 @@ class ServiceBusQueueTests(AzureMgmtTestCase):
     @RandomNameResourceGroupPreparer()
     @ServiceBusNamespacePreparer(name_prefix='servicebustest')
     @ServiceBusQueuePreparer(name_prefix='servicebustest', dead_lettering_on_message_expiration=True)
-    def test_github_issue_7079(self, servicebus_namespace_connection_string, servicebus_queue, **kwargs):
+    def test_receive_and_delete_reconnect_interaction(self, servicebus_namespace_connection_string, servicebus_queue, **kwargs):
+        # Note: This test was to guard against github issue 7079
         sb_client = ServiceBusClient.from_connection_string(
             servicebus_namespace_connection_string, debug=False)
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/tests/test_queues.py]: debug=False -> logging_enable=False

File: sdk/servicebus/azure-servicebus/tests/test_sessions.py [hunk 7]
```diff
@@ -1,943 +1,885 @@
...
+                for message in session:
+                    messages.append(message)
+                assert len(messages) == 0
+
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @RandomNameResourceGroupPreparer()
+    @ServiceBusNamespacePreparer(name_prefix='servicebustest')
+    @ServiceBusQueuePreparer(name_prefix='servicebustest', requires_session=True)
+    def test_session_by_session_client_conn_str_receive_handler_with_stop_TESTTEST(self, servicebus_namespace_connection_string, servicebus_queue, **kwargs):
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/tests/test_sessions.py]: (wondering the meaning of suffix "TESTTEST" here..)

File: sdk/servicebus/azure-servicebus/tests/test_sessions.py [hunk 8]
```diff
@@ -1,943 +1,885 @@
...
+                    assert message.lock_token
+                    assert not message.locked_until
+                    assert message._receiver
+                    with pytest.raises(TypeError):
+                        message.renew_lock()
+                    message.complete()
+
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @RandomNameResourceGroupPreparer(name_prefix='servicebustest')
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/tests/test_sessions.py]: we need to skip dead letter test in this file as well

File: sdk/servicebus/azure-servicebus/tests/test_sb_client.py [hunk 9]
```diff
@@ -145,35 +99,30 @@ def test_sb_client_readonly_credentials(self, servicebus_authorization_rule_conn
     @ServiceBusNamespaceAuthorizationRulePreparer(name_prefix='servicebustest', access_rights=[AccessRights.send])
     def test_sb_client_writeonly_credentials(self, servicebus_authorization_rule_connection_string, servicebus_queue, **kwargs):
```
Comment (yunhaoling) [inline sdk/servicebus/azure-servicebus/tests/test_sb_client.py]: we might want to skip this test first or tweak it a bit to bypass the issue in uamqp.
Same would apply to the readonly tests

File: sdk/servicebus/azure-servicebus/tests/test_sb_client.py (hunk 5 above)
Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/tests/test_sb_client.py]: Nope, that actually got added in a new PR I brought into T1 just recently, will be bringing that over when we "play catchup" on t1 changes that may not have gone into T2.

File: sdk/servicebus/azure-servicebus/tests/test_sessions.py (hunk 7 above)
Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/tests/test_sessions.py]: Hahahaha that's a silly debugging hack I use to not have to copy-paste the test method I'm debugging, so I can just hit scrollback to kick off the test from the command line.

It's been removed.

File: sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py (hunk 2 above)
Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/azure/servicebus/_servicebus_receiver.py]: It's in config now

File: sdk/servicebus/azure-servicebus/tests/test_sb_client.py (hunk 9 above)
Comment (KieranBrantnerMagee) [reply sdk/servicebus/azure-servicebus/tests/test_sb_client.py]: Thanks to your good work in sorting out UAMQP this should be resolved.

**Comment by KieranBrantnerMagee** [comment]
Comments resolved, migrating over to PR on branch that pulls in T1 test core changes.
https://github.com/yunhaoling/azure-sdk-for-python/pull/10

**Pull request closed without merging**